tests/integration/src/rust_masm_tests/debug_source_locations.rs

```diff
@@ -0,0 +1,167 @@
+//! Tests that verify debug source location information is correctly preserved
+//! from Rust source code through to MASM compilation and execution.
+//!
+
+use std::{
+    panic::{self, AssertUnwindSafe},
+    path::PathBuf,
+    process::Command,
+    sync::Arc,
+};
+
+use miden_core::Felt;
+use miden_debug::Executor;
+use miden_lib::MidenLib;
+use midenc_compile::compile_to_memory;
+use midenc_session::{InputFile, STDLIB};
+
+use crate::testing::setup;
+
+// Get path to examples/assert-debug-test test.
+fn get_assert_debug_test_path() -> PathBuf {
+    let manifest_dir = std::env::var("CARGO_MANIFEST_DIR")
+        .unwrap_or_else(|_| std::env::current_dir().unwrap().to_str().unwrap().to_string());
+    PathBuf::from(manifest_dir)
+        .parent()
+        .unwrap()
+        .parent()
+        .unwrap()
+        .join("examples")
+        .join("assert-debug-test")
+}
+
+fn create_executor_with_std(args: Vec<Felt>, package: &miden_mast_package::Package) -> Executor {
+    let mut exec = Executor::new(args);
+    let std_library = (*STDLIB).clone();
+    exec.dependency_resolver_mut()
+        .add(*std_library.digest(), std_library.clone().into());
+    let base_library = Arc::new(MidenLib::default().as_ref().clone());
+    exec.dependency_resolver_mut()
+        .add(*base_library.digest(), base_library.clone().into());
+    exec.with_dependencies(package.manifest.dependencies())
+        .expect("Failed to set up dependencies");
+    exec
+}
+
+#[test]
+fn test_rust_assert_macro_source_location_with_debug_executor() {
+    setup::enable_compiler_instrumentation();
+
+    let example_path = get_assert_debug_test_path();
+    let example_path_str = example_path.to_string_lossy();
+
+    let manifest_path = example_path.join("Cargo.toml");
+    let target_dir = example_path.join("target");
+    let status = Command::new("cargo")
+        .args([
+            "build",
+            "--release",
+            "--target",
+            "wasm32-unknown-unknown",
+            "--manifest-path",
+            manifest_path.to_str().unwrap(),
+            "--target-dir",
+            target_dir.to_str().unwrap(),
+        ])
+        .env_remove("CARGO_TARGET_DIR")
+        .status()
+        .expect("Failed to run cargo build");
+    assert!(status.success(), "Failed to build assert-debug-test example");
+
+    let wasm_path = example_path
+        .join("target")
+        .join("wasm32-unknown-unknown")
+        .join("release")
+        .join("assert_debug_test.wasm");
+
+    assert!(wasm_path.exists(), "WASM file not found at {wasm_path:?}.");
+
+    let input_file = InputFile::from_path(&wasm_path).expect("Failed to load wasm file");
+    let context = setup::default_context(
+        [input_file],
+        &[
+            "--debug",
+            "full",
+            &format!("-Ztrim-path-prefix={example_path_str}"),
```

Comment:
We should make `cargo-miden` use the `-Ztrim-path-prefix` option. I mean the test should call `cargo_miden::run` rather than use `cargo` with `midenc`.

The `cargo-miden` is the advised way of compiling the Miden package. We're not advertising the `cargo -> midenc` workflow that much, leaving it for people who want to tinker with the compiler. So `cargo-miden` should produce a Miden package with proper source location info.